### ISSUE TITLE
Training at 1280x720

### ISSUE THREAD

**Issue by miles-codes** [ultralytics/yolov5#6946]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi there,

I was wondering if it is possible to train at 1280x720 image shape? Obviously, network config and strides would have to be adjusted. train.py `imgsz` parameter is int, so currently training can only be done on square images.

I'm asking because my inference images are 1280x720, so I lose most of the time on preprocessing.

Thank you in advance :)

**Comment by glenn-jocher** [comment]
@miles-codes training settings are optimized for best results, I would leave them alone. If you want to train at native resolution then just do `python train.py --img 1280`.

YOLOv5 PyTorch model inference already exploits rectangular shapes, so if you run `python detect.py --img 1280` your image short side will automatically use 720.

**Comment by miles-codes** [comment]
I am aware of detect.py which uses letterbox to resize/pad input image to 1280x1280.
But my problem is that resizing/padding is slow, so I wanted to get rid of it by training on 1280x720 

Thanks for a quick response anyway :)

**Comment by glenn-jocher** [comment]
@miles-codes you're not understanding. If you pass a 1280x720 image to python detect.py --img 1280 then it **already by default exploits rectangular inference** for the minimum possible FLOPs.

**Comment by glenn-jocher** [comment]
@miles-codes also this is already clearly displayed in the console:

<img width="836" alt="Screenshot 2022-03-11 at 14 07 46" src="https://user-images.githubusercontent.com/26833433/157872322-1909c362-ab9e-4378-a63c-4f8e5beacf8e.png">

**Comment by miles-codes** [comment]
okay, thank you for the clarification :)

**Comment by HeChengHui** [comment]
@glenn-jocher 
if my detection input is 1920x1080 but my training images are of different sizes, is `--imgsze 1920 --rect` correct?

**Comment by glenn-jocher** [comment]
@HeChengHui if you want to detect at --img 1920 then train at --img 1920. It's very simple. Forget about any other options.

**Comment by HeChengHui** [comment]
@glenn-jocher Thank you!

A side thought, if detecting at 1920 is too slow and I changed detect size to 960, what would be the performance impact with a model trained at 1920?

**Comment by glenn-jocher** [comment]
@HeChengHui README Table curves shows models trained at 1280 running inference from 320 to 1280 image size.

<img width="846" alt="Screenshot 2022-03-15 at 12 06 05" src="https://user-images.githubusercontent.com/26833433/158364677-ef9bf83f-93ad-4034-925c-21e9df7cbac1.png">

**Comment by HeChengHui** [comment]
@glenn-jocher 
Is it correct to read it as each dot of each curve representing an inference at different img-size starting from 320 at the bottom and 1280 at the top?

**Comment by glenn-jocher** [comment]
@HeChengHui yes!

**Comment by HeChengHui** [comment]
@glenn-jocher Thank you for the guidance!

**Comment by wwcc1107** [comment]
if I want to detect at --img 1024 ,but train at --img 256，does it work?

**Comment by glenn-jocher** [comment]
@wwcc1107 depends on your definition of work

**Comment by stphtan94117** [comment]
i have a dataset from a lot of different img size.
their size are 640x480、1280x720 and 1920x1080 .
if i want to detect 1920x1080 size video, what do i set --img size in train.py ?
--imgsize 1920 is more better than 1280 or 640?
thanks.

**Comment by glenn-jocher** [comment]
@stphtan94117 if your objects are large then training at smaller images sizes will be fine, if they are small then larger images sizes up to native may be better. You can set any training --imgsz you want like this:
```
python train.py --imgsz 640
```
